<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <widget class="QLabel" name="cipherLable">
        <property name="text">
         <string>Cipher</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QRadioButton" name="caeserRadioButton">
          <property name="text">
           <string>Caeser</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="polybiusRadioButton">
          <property name="text">
           <string>Polybius</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="railfenceRadioButton">
          <property name="text">
           <string>Railfence</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="directTab">
       <attribute name="title">
        <string>Direct</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Cleartext</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="directClearText"/>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="directClearTextReadButton">
              <property name="text">
               <string>Read from file</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="directClearTextWriteButton">
              <property name="text">
               <string>Write to file</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QPushButton" name="directEncryptButton">
              <property name="text">
               <string>- Encrypt -&gt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="directDecryptButton">
              <property name="text">
               <string>&lt;- Decrypt -</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLineEdit" name="directKey">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>key</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>CipherText</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="directCipherText"/>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="directCipherTextReadButton">
              <property name="text">
               <string>Read from file</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="directCipherTextWriteButton">
              <property name="text">
               <string>Write to file</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="fileTab">
       <attribute name="title">
        <string>File</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_13">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string/>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_12">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Input</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QRadioButton" name="fileReadFile">
               <property name="text">
                <string>File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="fileReadPath">
               <property name="readOnly">
                <bool>true</bool>
               </property>
               <property name="placeholderText">
                <string>File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="fileReadButton">
               <property name="text">
                <string>Open...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QRadioButton" name="fileReadStdio">
               <property name="text">
                <string>Text</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="fileReadArea"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_19">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QPushButton" name="fileEncryptButton">
              <property name="text">
               <string>Encrypt</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="fileDecryptButton">
              <property name="text">
               <string>Decrypt</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLineEdit" name="fileKey">
            <property name="placeholderText">
             <string>key</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string/>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Output</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QRadioButton" name="fileWriteFile">
               <property name="text">
                <string>File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="fileWritePath">
               <property name="readOnly">
                <bool>true</bool>
               </property>
               <property name="placeholderText">
                <string>File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="fileWriteButton">
               <property name="text">
                <string>Open...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QRadioButton" name="fileWriteStdio">
               <property name="text">
                <string>Text</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="fileWriteArea"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
